service/src/test/java/org/sunbird/learner/actors/coursemanagement/CourseBatchManagementActorTest.java

```diff
@@ -0,0 +1,355 @@
+  }
+
+  // course Ended
+  @Test
+  public void testUpdateCompletedCourseBatchFailureWithEndDate() {
+
+    int courseStatus = ProjectUtil.ProgressStatus.COMPLETED.getValue();
+    Response mockCassandraResponse = getMockCassandraRecordByIdResponse(courseStatus);
+    ProjectCommonException ex = updateCourseBatchFailure(null, endDate, mockCassandraResponse);
+    Assert.assertTrue(
+        ((ProjectCommonException) ex)
+            .getCode()
+            .equals(ResponseCode.courseBatchEndDateError.getErrorCode()));
+  }
+
+  @Test
+  public void testUpdateCompletedCourseBatchFailureWithStartDate() {
+
+    int courseStatus = ProjectUtil.ProgressStatus.COMPLETED.getValue();
+    Response mockCassandraResponse = getMockCassandraRecordByIdResponse(courseStatus);
+    ProjectCommonException ex = updateCourseBatchFailure(startDate, null, mockCassandraResponse);
+    Assert.assertTrue(
+        ((ProjectCommonException) ex)
+            .getCode()
+            .equals(ResponseCode.invalidBatchStartDateError.getErrorCode()));
+  }
+
+  @Test
+  public void testUpdateCompletedCourseBatchFailureWithStartDateAndEndDate() {
+
+    int courseStatus = ProjectUtil.ProgressStatus.COMPLETED.getValue();
+    Response mockCassandraResponse = getMockCassandraRecordByIdResponse(courseStatus);
+    ProjectCommonException ex = updateCourseBatchFailure(startDate, endDate, mockCassandraResponse);
+    Assert.assertTrue(
+        ((ProjectCommonException) ex)
+            .getCode()
+            .equals(ResponseCode.invalidBatchStartDateError.getErrorCode()));
+  }
+
+  private Response getMockCassandraResult() {
+
+    Response response = new Response();
+    response.put("response", "SUCCESS");
+    return response;
+  }
+
+  private Response getMockCassandraRecordByIdResponse(int courseStatus) {
+
+    String existingStartDate = "";
+    String existingEndDate = "";
+
+    Response response = new Response();
+    List<Map<String, Object>> list = new ArrayList<>();
+    Map<String, Object> courseResponseMap = new HashMap<>();
+
+    courseResponseMap.put(JsonKey.ID, BATCH_ID);
+    courseResponseMap.put(JsonKey.VER, "v1");
+    courseResponseMap.put(JsonKey.NAME, BATCH_NAME);
+    courseResponseMap.put(JsonKey.COUNTER_INCREMENT_STATUS, "false");
+    courseResponseMap.put(JsonKey.ENROLMENTTYPE, "Invite-only");
+    courseResponseMap.put(JsonKey.COURSE_ID, "do_2125635353836584961866");
+    courseResponseMap.put(JsonKey.COURSE_CREATED_FOR, new ArrayList<Object>());
+    courseResponseMap.put(JsonKey.STATUS, courseStatus);
+
+    Calendar calendarStartDate = Calendar.getInstance();
+    Calendar calendarEndDate = Calendar.getInstance();
+
+    if (courseStatus == ProjectUtil.ProgressStatus.COMPLETED.getValue()) {
+      calendarStartDate.add(Calendar.DAY_OF_MONTH, -4);
+      calendarEndDate.add(Calendar.DAY_OF_MONTH, 3);
+
+      samePastStartDate = existingStartDate = format.format(calendarStartDate.getTime());
```

Comment:
Make existingStartDate and existingEndDate instance variables and samePast* variables as local.